ui/src/error.ts

```diff
@@ -2,22 +2,48 @@ import * as notification from "./notification";
 import * as ipc from "./ipc";
 import * as svelteStore from "svelte/store";
 
-export enum Variant {
-  EntityExists = "ENTITY_EXISTS",
-  GitError = "GIT_ERROR",
-  NotFound = "NOT_FOUND",
-}
-
 export interface Error {
+  // A code in kebab-case for easy identification of the error.
```

Comment:
Why this formulation for a code, out of interest?

Reply:
Are you referring to “kebab-case”. I just chose this because it is easiest to type compared to screaming snake case or others.

Reply:
Ya, so I'd prefer to have some structured way of building an error code. For example, still have a string but have it be `E0001` and increment the codes. As opposed to people picking random sequences of words :smile: 

What are your thoughts?

Reply:
I’m hesitant to use number codes. The primary purpose of these codes is to help developers identitfy an error when a user requests support or we see it during development. If we use descriptive strings as error codes it becomes much easier to identify the error without having to refer to the lookup from number to error description. Conversely, I don’t see an advantage of using numbers as error codes. We would still need to choose an adequate sequence of words to describe the error codes.

Reply:
> I’m hesitant to use number codes. The primary purpose of these codes is to help developers identitfy an error when a user requests support or we see it during development.

Me too. My main goal from this is to figure out how to make this not ad-hoc process. I can see errors happening like mispellings, copy-and-paste without changing the code, or it's the same error but gets copy-pasted and they have the potential to go out of sync.

> We would still need to choose an adequate sequence of words to describe the error codes.

Ya, I would say reasons and codes should be paired with each other.

In the description you said:
> I’ve decided not to make code an enum. Having a centralized place that lists all errors does not help us with debugging. When debugging we need to look at the place where the error originated which can easily be found be grepping for the error code.

What about having constants in a file, so something like: `const PROJECT_FAILED = name => { code: "project-request-failure", reason: The project {name} could not be loaded" }`?

We can then grep for `PROJECT_FAILED` to see where it's used.

Reply:
Resolved in https://github.com/radicle-dev/radicle-upstream/pull/1221#issuecomment-725958625